gpcontrib/arenadata_toolkit/expected/adb_relation_storage_size_test.out

```diff
@@ -0,0 +1,95 @@
+CREATE TABLE ao_table_with_toast(a INT, b TEXT)
+WITH (APPENDOPTIMIZED=true, ORIENTATION=COLUMN)
+DISTRIBUTED BY (a);
+CREATE TABLE ao_table_without_toast(a INT, b INT)
+WITH (APPENDOPTIMIZED=true, ORIENTATION=COLUMN)
+DISTRIBUTED BY (a);
+-- Check that toast exists only for "with_toast" tables
+SELECT relname, reltoastrelid != 0 with_toast
+FROM pg_class
+WHERE relname IN ('heap_table_with_toast', 'heap_table_without_toast',
+                  'ao_table_with_toast', 'ao_table_without_toast')
+ORDER BY 1;
+         relname          | with_toast 
+--------------------------+------------
+ ao_table_with_toast      | t
+ ao_table_without_toast   | f
+ heap_table_with_toast    | t
+ heap_table_without_toast | f
+(4 rows)
+
+-- Insert initial data to tables
+INSERT INTO heap_table_with_toast SELECT i, 'short_text' FROM generate_series(1,15) AS i;
+INSERT INTO heap_table_without_toast SELECT i, i*10 FROM generate_series(1,15) AS i;
+INSERT INTO ao_table_with_toast SELECT i, 'short_text' FROM generate_series(1,15) AS i;
+INSERT INTO ao_table_without_toast SELECT i, i*10 FROM generate_series(1,15) AS i;
+-- Check sizes on segments
+SELECT relname, sizes.gp_segment_id, sizes.size
+FROM pg_class, arenadata_toolkit.adb_relation_storage_size_on_segments(oid) sizes
+WHERE relname IN ('heap_table_with_toast', 'heap_table_without_toast',
+                  'ao_table_with_toast', 'ao_table_without_toast')
+ORDER BY 1, 2;
+         relname          | gp_segment_id | size  
+--------------------------+---------------+-------
+ ao_table_with_toast      |             0 |   160
+ ao_table_with_toast      |             1 |   136
+ ao_table_with_toast      |             2 |   192
+ ao_table_without_toast   |             0 |   128
+ ao_table_without_toast   |             1 |   112
+ ao_table_without_toast   |             2 |   144
+ heap_table_with_toast    |             0 | 32768
+ heap_table_with_toast    |             1 | 32768
+ heap_table_with_toast    |             2 | 32768
+ heap_table_without_toast |             0 | 32768
+ heap_table_without_toast |             1 | 32768
+ heap_table_without_toast |             2 | 32768
+(12 rows)
+
+-- Add random large data to get non-zero toast table's size
+UPDATE heap_table_with_toast SET b = (
+    SELECT string_agg( chr(trunc(65+random()*26)::integer), '')
+    FROM generate_series(1,50000))
+WHERE a = 1;
+UPDATE ao_table_with_toast SET b = (
+    SELECT string_agg( chr(trunc(65+random()*26)::integer), '')
+    FROM generate_series(1,50000))
+WHERE a = 1;
+SELECT relname, sizes.gp_segment_id, sizes.size
+FROM pg_class, arenadata_toolkit.adb_relation_storage_size_on_segments(oid) sizes
+WHERE relname IN ('heap_table_with_toast', 'ao_table_with_toast')
+ORDER BY 1, 2;
+        relname        | gp_segment_id | size  
+-----------------------+---------------+-------
+ ao_table_with_toast   |             0 |   160
+ ao_table_with_toast   |             1 | 50248
+ ao_table_with_toast   |             2 |   192
```

Comment:
@red1452 , Test without patch shows the same size of segment files for AO tables.  It seems like the toast relation with `reltoastrelid` does not even exist in base directory.  Why is it happening? How the TOAST mechanism works for AO tables?

Reply:
At this test, I add one row to toast table only at one segment. Empty toast table has zero size. Without patch sizes on segments will be the same as sizes before insert large value to this table.

Reply:
Without and with patch the size of all segment files is the same after inserting large value.

Reply:
Anyway, toast table has zero size for the column with large value

Reply:
```
CREATE TABLE ao_table_with_toast(a INT, b TEXT)
WITH (APPENDOPTIMIZED=true, ORIENTATION=ROW);

insert into ao_table_with_toast select 1, (SELECT string_agg( chr(trunc(65+random()*26)::integer), '') FROM generate_series(1,50000));

SELECT oid::regclass AS Table,
       reltoastrelid::regclass As TOAST_Table,
       pg_relation_size(reltoastrelid) AS toast_size
FROM pg_class
WHERE relkind = 'r'
  AND reltoastrelid <> 0
ORDER BY 3 DESC; 
                  table                    |       toast_table       | toast_size 
--------------------------------------------+-------------------------+------------
ao_table_with_toast                        | pg_toast.pg_toast_16884 |      65536
```

Reply:
**ORIENTATION=ROW** works

Reply:
You are right, fixed.